exporter/redis.go

```diff
@@ -282,104 +227,146 @@ func parseKeyArg(keysArgString string) (keys []dbKeyPair, err error) {
-			Namespace: namespace,
-			Name:      "key_value",
-			Help:      "The value of \"key\"",
-		}, []string{"addr", "alias", "db", "key"}),
-		keySizes: prometheus.NewGaugeVec(prometheus.GaugeOpts{
-			Namespace: namespace,
-			Name:      "key_size",
-			Help:      "The length or size of \"key\"",
-		}, []string{"addr", "alias", "db", "key"}),
-		scriptValues: prometheus.NewGaugeVec(prometheus.GaugeOpts{
-			Namespace: namespace,
-			Name:      "script_value",
-			Help:      "Values returned by the collect script",
-		}, []string{"addr", "alias", "key"}),
-		duration: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: namespace,
-			Name:      "exporter_last_scrape_duration_seconds",
-			Help:      "The last scrape duration.",
-		}),
+		redisAddr: redisURI,
+		options:   opts,
+		namespace: opts.Namespace,
+
 		totalScrapes: prometheus.NewCounter(prometheus.CounterOpts{
-			Namespace: namespace,
+			Namespace: opts.Namespace,
 			Name:      "exporter_scrapes_total",
 			Help:      "Current total redis scrapes.",
 		}),
-		scrapeErrors: prometheus.NewGauge(prometheus.GaugeOpts{
-			Namespace: namespace,
-			Name:      "exporter_last_scrape_error",
-			Help:      "The last scrape error status.",
-		}),
+
+		targetScrapeDuration: prometheus.NewSummary(
+			prometheus.SummaryOpts{
+				Namespace: opts.Namespace,
+				Name:      "target_scrape_collection_duration_seconds",
+				Help:      "Duration of collections by the SNMP exporter",
+			},
+		),
+		targetScrapeRequestErrors: prometheus.NewCounter(
+			prometheus.CounterOpts{
+				Namespace: opts.Namespace,
+				Name:      "target_scrape_request_errors_total",
+				Help:      "Errors in requests to the SNMP exporter",
+			},
+		),
+	}
+
+	if e.options.ConfigCommandName == "" {
+		e.options.ConfigCommandName = "CONFIG"
 	}
 
 	var err error
 
-	if e.keys, err = parseKeyArg(checkKeys); err != nil {
+	if e.keys, err = parseKeyArg(opts.CheckKeys); err != nil {
 		return &e, fmt.Errorf("Couldn't parse check-keys: %#v", err)
 	}
 	log.Debugf("keys: %#v", e.keys)
 
-	if e.singleKeys, err = parseKeyArg(checkSingleKeys); err != nil {
+	if e.singleKeys, err = parseKeyArg(opts.CheckSingleKeys); err != nil {
 		return &e, fmt.Errorf("Couldn't parse check-single-keys: %#v", err)
 	}
 	log.Debugf("singleKeys: %#v", e.singleKeys)
 
-	e.initGauges()
+	if opts.IncludeMetricTotalSysMemory {
+		metricMapGauges["total_system_memory"] = "total_system_memory_bytes"
+	}
+
+	e.metricDescriptions = map[string]*prometheus.Desc{}
+	e.metricDescriptions["up"] = newMetricDescr(opts.Namespace, "up", "Information about the Redis instance", nil)
+	e.metricDescriptions["instance_info"] = newMetricDescr(opts.Namespace, "instance_info", "Information about the Redis instance", []string{"role", "redis_version", "redis_build_id", "redis_mode", "os"})
+	e.metricDescriptions["last_scrape_duration"] = newMetricDescr(opts.Namespace, "exporter_last_scrape_duration_seconds", "The last scrape duration", nil)
```

Comment:
You'd generally provide information about this scrape.

Reply:
Ok, moved that up to the exporter level and made it a summary.